scipy/linalg/tests/test_lapack.py

```diff
@@ -650,6 +651,47 @@ def test_invalid_matrix_shapes(self, ldab, n, ldb, nrhs):
         assert_raises(Exception, tbtrs, ab, b)
 
 
+
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('norm', ['I', '1', 'O'])
+@pytest.mark.parametrize('uplo', ['U', 'L'])
+@pytest.mark.parametrize('diag', ['N', 'U'])
+@pytest.mark.parametrize('n', [3, 10])
+def test_trcon(dtype, norm, uplo, diag, n):
+    # Simple way to get deterministic (unlike `hash`) integer seed based on arguments
+    random.seed(f"{dtype}{norm}{uplo}{diag}{n}")
+    rng = np.random.default_rng(random.randint(0, 9999999999999))
+
+    A = rng.random(size=(n, n)) + rng.random(size=(n, n))*1j
+    # make the condition numbers more interesting
+    offset = rng.permuted(np.logspace(0, rng.integers(0, 10), n))
```

Comment:
Lazy way of getting more widely varying condition numbers. I considered generating orthogonal matrices and specifying the eigenvalues, but I think this works?

Reply:
I understand the temptation here but we really don't need to test functionality of the routine. That's LAPACK folks' business. We just need to make sure the wrappers are functional which we can do by just comparing to `1 / np.linalg.cond(A, 1)` for any triangular `A` and they are within factor of 2 to 5 either way. 

I'd choose to keep the wrapper tests light since this file is getting out of control already.

Reply:
It's not testing the functionality of the routine. It is just testing the plumbing. If we test for a single matrix, it could easily be a coincidence when the tolerance is so loose.

Reply:
I think it's not that much of importance I initially mentioned here. So if you are happy with it, I'm OK with it. 

If you want to test specific arrays, I made a mini cython script to test different flavors that works in a Jupyter notebook cell. Just run `%load_ext cython` on a prior cell somewhere and then the following should run and give you the results of each flavor. You can also play around with the arrays if you are interested or use it for other purposes

```cython

%%cython -f
import numpy as np
cimport numpy as cnp
from scipy.linalg.cython_lapack cimport strcon, dtrcon, ctrcon, ztrcon

cdef double rcondd
cdef float rcondf
cdef int n = 10
cdef int info
cdef float* Rfd
cdef float* workfd
cdef double* Rdd
cdef double* workdd
cdef float complex* Rcfd
cdef float complex* workcfd
cdef double complex* Rcdd
cdef double complex* workcdd

cdef int* irworkd
rng = np.random.default_rng(12345678987654321)
irwork = np.empty(n, dtype=np.int32)
irworkd = <int*>cnp.PyArray_DATA(irwork)
print("\n")

# ---- S - np.float32
Rf = rng.integers(low=-n, high=n, size=(n, n)).astype(np.float32)
Rfd = <float*>cnp.PyArray_DATA(Rf)
workf = np.empty(3*n, dtype=np.float32)
workfd = <float*>cnp.PyArray_DATA(workf)

# Note: C array is Fortran order transposed
strcon(<char*>"O",<char*>"L",<char*>"N", &n, Rfd, &n, &rcondf, workfd, irworkd, &info)
print(f"STRCON norm=1, uplo=L, {info=}, {rcondf=}")
strcon(<char*>"I",<char*>"U",<char*>"N", &n, Rfd, &n, &rcondf, workfd, irworkd, &info)
print(f"STRCON norm=I, uplo=U, {info=}, {rcondf=}")

# ---- D - np.float64
Rd = rng.integers(low=-n, high=n, size=[n, n]).astype(np.float64)
Rdd = <double*>cnp.PyArray_DATA(Rd)
workd = np.empty(3*n, dtype=np.float64)
workdd = <double*>cnp.PyArray_DATA(workd)

dtrcon(<char*>"1",<char*>"L",<char*>"N", &n, Rdd, &n, &rcondd, workdd, irworkd, &info)
print(f"DTRCON {info=}, {rcondd=}")

# ---- C - np.complex64
Rcf = rng.standard_exponential(size=(n, n), dtype=np.float32) + rng.uniform(size=(n, n)).astype(np.float32)*np.complex64(1.j)
Rcfd = <float complex*>cnp.PyArray_DATA(Rcf)
workcf = np.empty(2*n, dtype=np.complex64)
workcfd = <float complex*>cnp.PyArray_DATA(workcf)
# Recycle workf here for rwork
ctrcon(<char*>"1",<char*>"L",<char*>"N", &n, Rcfd, &n, &rcondf, workcfd, workfd, &info)
print(f"CTRCON {info=}, {rcondf=}")


# ---- Z - np.complex128
Rcd = rng.standard_exponential(size=(n, n)) + rng.uniform(size=(n, n))*1.j
Rcdd = <double complex*>cnp.PyArray_DATA(Rcd)
workcd = np.empty(2*n, dtype=np.complex64)
workcdd = <double complex*>cnp.PyArray_DATA(workcd)
ztrcon(<char*>"1",<char*>"L",<char*>"N", &n, Rcdd, &n, &rcondd, workcdd, workdd, &info)
print(f"ZTRCON {info=}, {rcondd=}")



```

Reply:
Ah great, yeah, I was going to try it with Cython if you were concerned with the disagreement. But yeah, I'm happy with it as-is is now. I will play it with Cython later and can always open a new PR if there is something wrong. Thanks!